index.js

```diff
@@ -45,59 +46,64 @@ export function *iter(source) {
     // we consume at most one col per outer loop
     if (sourceCharCodeAt() == C_NEWLINE) {
       // yielding row and resetting is smaller but about 10% slower
-      yield row.splice(0, row.length);
+      yield row.splice(0);
       ++i;
     }
 
-    if (i >= length) {
+    if (!(i < length)) {
       break;
     }
 
-    if (i > newline) {
-      newline = nextIndex('\n');
```

Comment:
Don't we need this?

Reply:
Yes... but it's now in-lined when `newline` is used. You don't actually need to update `newline` every loop, as it's only used in one place.